test/test-Benchmark.js

```diff
@@ -0,0 +1,129 @@
+/* globals require, exports, XPCOMUtils, Task, Services, ClientID */
+"use strict";
+
+const {before, after} = require("sdk/test/utils");
+const tabs = require("sdk/tabs");
+const simplePrefs = require("sdk/simple-prefs");
+const {Cu} = require("chrome");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/ClientID.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "Task",
+                                  "resource://gre/modules/Task.jsm");
+
+const {Benchmark} = require("./lib/Benchmark.js");
+const {getTestActivityStream} = require("./lib/utils");
+
+let app;
+let ACTIVITY_STREAMS_URL;
+
+const roundsUpTest = [
+  [0, 1],
+  [1, 1],
+  [2, 2],
+  [3, 3],
+  [5, 5],
+  [9, 10],
+  [99, 100],
+  [100, 100],
+  [140, 200],
+  [170, 200],
+  [270, 300],
+  [499, 500],
+  [500, 500],
+  [501, 1000]
+];
+
+exports.test_roundsUp = function(assert) {
+  let bench = new Benchmark(b => {});
+  for (let [n, expect] of roundsUpTest) {
+    const actual = bench._roundsUp(n);
+    assert.equal(expect, actual);
+  }
+};
+
+function fib(n) {
+  if (n < 2) {
+    return n;
+  }
+
+  return fib(n - 1) + fib(n - 2);
+}
+
+exports.test_benchmark_fib = function*(assert) {
+  let bench = new Benchmark(b => {
+    for (let i = 0; i < b.N; i++) {
+      fib(20);
+    }
+  });
+  const result = yield bench.launch();
+  assert.ok(result.ok, "it should run the benchmark and report the status");
+  assert.equal(bench._usPerOp(), result.usPerOp);
+};
+
+const openActivityStream = Task.async(function*(openUrl) {
```

Comment:
I have a suspicion that this measures all of openActivityStream, as well as some of closeActivityStream.  The reason is that even though the code doesn't wait for the tab to close before starting the next open, the closing will eat some of the cycles on the UI thread (and, in the e10s case, in the content process). 
So the wall clock time measure is likely to in part reflect that the event queue on both of those threads spent time away from the opening task.

 In the (future) case of multiple content processes, this may or may not get diluted more, depending on how the tabs are sorted across content processes.

It might make sense to explicitly do and wait for teardown completion with the timer turned off before starting the next iteration of the test.

Not closing the tabs until the whole test finishes is another option, though various caching strategies (eg XUL "brutal sharing") will kick in that case.